writing/example.tex

```diff
@@ -0,0 +1,203 @@
+    language=slim,
+    backgroundcolor=\color{backcolour},   
+    commentstyle=\color{slimcomment},
+    keywordstyle=[1]\color{slimkeyword},
+    keywordstyle=[2]\color{slimstage},
+    keywordstyle=[3]\color{slimidentifier},
+    numberstyle=\tiny\color{codegray},
+    stringstyle=\color{slimstring},
+    basicstyle=\ttfamily\footnotesize,
+    escapeinside={*@}{@*},
+    breakatwhitespace=false,         
+    breaklines=true,                 
+    captionpos=b,                    
+    keepspaces=true,                 
+    numbers=left,                    
+    numbersep=2pt,                  
+    showspaces=false,                
+    showstringspaces=false,
+    showtabs=false,                  
+    tabsize=2
+}
+\lstset{style=slimstyle}
+
+\title{SLiM code in \LaTeX}
+
+\begin{document}
+
+Here is an example SLiM script.
+It's not the simplest, but it does demonstrate a number of different language features
+to be syntax highlighted.
+
+Note that below we include labels on particular lines,
+so we can refer to them.
+For instance,
+at line~\ref{setup} we use the function defined later,
+at line~\ref{function}.
+The way this works is that the characeters \verb|*@| and \verb|@*| delimit
+code that is ``escaped'' to \LaTeX.
+
+\begin{lstlisting}{language=slim}
+initialize() {
+	initializeSLiMModelType("nonWF");
+	initializeSLiMOptions(dimensionality="xy");
+	
+	defaults = Dictionary(
+		"SEED", getSeed(),
+		"SD", 0.3,         // sigma_D, dispersal distance
+		"SX", 0.3,         // sigma_X, interaction distance for measuring local density
+		"SM", 0.3,         // sigma_M, mate choice distance
+		"K", 5,            // carrying capacity per unit area
+		"LIFETIME", 4,     // average life span
+		"WIDTH", 25.0,     // width of the simulated area
+		"HEIGHT", 25.0,    // height of the simulated area
+		"RUNTIME", 200,    // total number of ticks to run the simulation for
+		"L", 1e8,          // genome length
+		"R", 1e-8,         // recombination rate
+		"MU", 0            // mutation rate
+		);	
+	
+	// Set up parameters with a user-defined function
+    setupParams(defaults); *@ \label{setup} @*
+	
+	// Set up constants that depend on externally defined parameters
+	defineConstant("FECUN", 1 / LIFETIME);
+	defineConstant("RHO", FECUN / ((1 + FECUN) * K));
+	defineConstant("PARAMS", defaults);
+	
+	setSeed(SEED);
+	
+	// basic neutral genetics
+	initializeMutationRate(MU);
+	initializeMutationType("m1", 0.5, "f", 0.0);
+	initializeGenomicElementType("g1", m1, 1.0);
+	initializeGenomicElement(g1, 0, L-1);
+	initializeRecombinationRate(R);
+	
+	// spatial interaction for local density measurement
+	initializeInteractionType(1, "xy", reciprocal=T, maxDistance=3 * SX);
+	i1.setInteractionFunction("n", 1, SX);
+	
+	// spatial interaction for mate choice
+	initializeInteractionType(2, "xy", reciprocal=T, maxDistance=3 * SM);
+	i2.setInteractionFunction("n", 1, SM);
+}
+
+1 first() {
+	sim.addSubpop("p1", asInteger(K * WIDTH * HEIGHT));
```

Comment:
Is there any way to express line-break preferences?  If so, that'd be nice to show here.  In the PDF, the "1 first() {" line is on page 1 and then the rest of the code for it is on page 2; what I'd like to see demonstrated here (if it is easy) is a way to tell LaTeX "keep these lines together" as a hint to the pagination algorithm.  But if you're tired of working on this PR, I get that too.  :->

Reply:
Well, I am tired of working on the PR and I don't think that'll be easy...

Reply:
ok, no worries, just thought it might be trivial in TeX